compliance/utils/services/github.py

```diff
@@ -413,6 +414,44 @@ def get_commit_details(self, repo, since, branch='master', path=None):
             opts['path'] = path
         return self._make_request('get', f'repos/{repo}/commits', params=opts)
 
+    def get_pull_request_details(
+        self, repo, since, base='master', state='all', sort='created'
+    ):
```

Comment:
I don't think that we should be changing much of the default behavior of the API here.  Why did you choose to limit the options for this to base, state, and sort (leaving off direction)?

I suggest changing the name of the method to `get_pul_requests`.  `get_pull_request_details` suggests getting details for a single pull request.  Also, I think that changing the method signature to `def get_pull_requests(self, repo, since=None, **kwargs):` would allow the method to adhere better to the API options while still providing internal pagination and "since" logic.  If a "since" date is provided then you should force `sort` to be `created`.  Right now if I provide a "since" date but sort by "updated" or "popularity" then this method could produce odd results.

Reply:
Thank you for your feedback. I totally agree with your suggestion. I work on this and update this PR soon.